include/aws/crt/Optional.h

```diff
@@ -65,19 +66,33 @@ namespace Aws
                 return *this;
             }
 
-            Optional(const Optional<T> &other) = default;
+            Optional(const Optional<T> &other)
+            {
+                if (other.m_value)
+                {
+                    new (&m_storage) T(*other.m_value);
+                    m_value = reinterpret_cast<T *>(&m_storage);
+                }
+                else
+                {
+                    m_value = nullptr;
+                }
+            }
 
-            Optional(Optional<T> &&other)
+            Optional(Optional<T> &&other) noexcept
```

Comment:
This shouldn't always be noexcept

from [cppreference](https://en.cppreference.com/w/cpp/utility/optional/optional)
> Throws any exception thrown by the constructor of T. Has the following noexcept specification:  
> noexcept(std::is_nothrow_move_constructible<T>::value)

I have no idea how to write that in template-ese

Reply:
clang-tidy told me move constructors must always be noexcept.... is the argument that this may not be a move since it's a forwarding reference?